lib/read.js

```diff
@@ -133,39 +139,6 @@ function read (req, res, next, parse, debug, options) {
   })
 }
 
-/**
- * Get the content stream of the request.
- *
- * @param {object} req
- * @param {function} debug
- * @param {boolean} [inflate=true]
- * @return {object}
- * @api private
- */
-
-function contentstream (req, debug, inflate) {
-  var encoding = (req.headers['content-encoding'] || 'identity').toLowerCase()
-  var length = req.headers['content-length']
-
-  debug('content-encoding "%s"', encoding)
-
-  if (inflate === false && encoding !== 'identity') {
-    throw createError(415, 'content encoding unsupported', {
-      encoding: encoding,
-      type: 'encoding.unsupported'
-    })
-  }
-
-  if (encoding === 'identity') {
-    req.length = length
-    return req
-  }
-
-  var stream = createDecompressionStream(encoding, debug)
-  req.pipe(stream)
-  return stream
-}
```

Comment:
This can be seen as follow-up to #564. I included the `contentstream` function in the main `read` function for multiple reasons:
- to prevent the property `length` from being set and reset on the `req` object. 
- This also improves code by only setting the `raw-body` `length` option and thus also only calling `req.headers['content-length']` when the requests needs no inflation.